src/utils.js

```diff
@@ -55,34 +58,16 @@ export function merge () {
   return copy
 }
 
-export function backfill (subject, properties) {
-  if (subject == null) {
-    return properties
-  }
-
-  let copy = subject
-
-  for (var key in properties) {
-    if (copy.hasOwnProperty(key) === false) {
-      if (copy === subject) {
-        copy = clone(subject)
-      }
-
-      copy[key] = properties[key]
-    }
-  }
-
-  return copy
-}
-
 /**
  * Basic prototypal inheritence
  */
 export function inherit (Child, Ancestor, proto) {
-  Child.__proto__ = Ancestor
+  if (Ancestor) {
+    Child.__proto__ = Ancestor
+  }
 
   Child.prototype = merge(Object.create(Ancestor.prototype), {
-    constructor: Child
+    constructor: Child.prototype.constructor
   }, proto)
```

Comment:
I changed this to be more inline with how Buble (buble.surge.sh) transpiles classes. I'm starting to wonder if we should just use classes again. They are much nicer to work with. 

Thoughts @greypants @mackermedia ?